projects/timer_hw/src/main.c

```diff
@@ -0,0 +1,51 @@
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+#define COUNTER_PERIOD_MS 500
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+void prv_timer_callback_2(SoftTimerId timer_id, void *context);
+
+void prv_timer_callback(SoftTimerId timer_id, void *context) {
```

Comment:
You can (and should) do all the soft-timer functionality in one timer callback function instead of 2, see if you can figure it out, but if not feel free to ask for help 🙂

Reply:
Additionally, all private functions in a file should be declared as static, ie `static void prv_*`